lib/tool-bar-view.js

```diff
@@ -0,0 +1,229 @@
+        atom.config.set('tool-bar.position', 'Bottom');
+      }),
+      atom.commands.add('atom-workspace', 'tool-bar:position-left', () => {
+        this.updatePosition('Left');
+        atom.config.set('tool-bar.position', 'Left');
+      }),
+      atom.config.observe('tool-bar.iconSize', newValue => {
+        this.updateSize(newValue);
+      }),
+      atom.config.onDidChange('tool-bar.position', ({newValue, oldValue}) => {
+        if (atom.config.get('tool-bar.visible')) {
+          this.show();
+        }
+      }),
+      atom.config.onDidChange('tool-bar.visible', ({newValue, oldValue}) => {
+        if (newValue) {
+          this.show();
+        } else {
+          this.hide();
+        }
+      })
+    );
+
+    if (supportFullWidth) {
+      this.subscriptions.add(
+        atom.config.onDidChange('tool-bar.fullWidth', ({newValue, oldValue}) => {
+          if (atom.config.get('tool-bar.visible')) {
+            this.show();
+          }
+        })
+      );
+    }
+
+    if (atom.config.get('tool-bar.visible')) {
+      this.show();
+    }
+
+    this.drawGutter = this.drawGutter.bind(this);
+
+    this.on('scroll', this.drawGutter);
+    window.addEventListener('resize', this.drawGutter);
+  }
+
+  addItem (newItem) {
+    const lastItem = this.items.filter(item =>
+      item.group !== newItem.group).pop();
+
+    if (newItem.priority == null) {
+      newItem.priority = lastItem ? lastItem.priority + 1 : 50;
+    }
+
+    if (atom.devMode) {
+      newItem.get(0).dataset.group = newItem.group;
+      newItem.get(0).dataset.priority = newItem.priority;
+    }
+
+    const index = this.items.findIndex(existingItem =>
+      (existingItem.priority > newItem.priority));
+    const nextItem = this.items[index];
+
+    this.items.splice(index, 0, newItem);
+
+    const newElement = atom.views.getView(newItem);
+    const nextElement = atom.views.getView(nextItem);
+    this.element.insertBefore(newElement, nextElement);
+
+    this.drawGutter();
+
+    return nextItem;
+  }
+
+  removeItem (item) {
+    item.destroy();
+    this.items.splice(this.items.indexOf(item), 1);
+    this.drawGutter();
+  }
+
+  destroy () {
+    this.items.forEach(item => item.destroy());
+    this.items = null;
+
+    this.subscriptions.dispose();
+    this.subscriptions = null;
+
+    this.hide();
+    this.remove();
+    window.removeEventListener('resize', this.drawGutter);
+
+    this.emitter.emit('did-destroy');
+    this.emitter.dispose();
+    this.emitter = null;
+  }
+
+  updateSize (size) {
+    this.removeClass('tool-bar-12px tool-bar-16px tool-bar-24px tool-bar-32px');
+    this.addClass(`tool-bar-${size}`);
+  }
+
+  updatePosition (position) {
+    this.removeClass(
+      'tool-bar-top tool-bar-right tool-bar-bottom tool-bar-left ' +
+      'tool-bar-horizontal tool-bar-vertical'
+    );
+
+    const fullWidth = supportFullWidth && atom.config.get('tool-bar.fullWidth');
+
+    if (position === 'Top') {
```

Comment:
Done! I tried to mimic the short if-else syntax with ternaries. Let me know if you prefer if-else too.